server/session/handler_item_stack_request.go

```diff
@@ -343,31 +408,35 @@ func (h *ItemStackRequestHandler) verifySlot(slot protocol.StackRequestSlotInfo,
 	if id := item_id(i); id != clientID {
 		return fmt.Errorf("stack ID mismatch: client expected %v, but server had %v", clientID, id)
 	}
-	inventory, _ := s.invByID(int32(slot.ContainerID))
 
+	inv, _ := s.invByID(int32(slot.ContainerID))
 	sl := int(slot.Slot)
-	if inventory == s.offHand {
+	if inv == s.offHand {
 		sl = 0
 	}
 
-	if inventory.SlotLocked(sl) {
-		return fmt.Errorf("slot in inventory was locked")
+	if inv.SlotLocked(sl) {
+		return fmt.Errorf("slot in inv was locked")
 	}
 	return nil
 }
 
 // resolveID resolves the stack network ID in the slot passed. If it is negative, it points to an earlier
 // request, in which case it will look it up in the changes of an earlier response to a request to find the
 // actual stack network ID in the slot. If it is positive, the ID will be returned again.
-func (h *ItemStackRequestHandler) resolveID(slot protocol.StackRequestSlotInfo) (int32, error) {
+func (h *ItemStackRequestHandler) resolveID(s *Session, slot protocol.StackRequestSlotInfo) (int32, error) {
```

Comment:
This change is a bit pointless. Just move the inventory validity check up a bit in your crafting code so that this isn't necessary.